### PR TITLE
User group#10

### PR DESCRIPTION
**Pull request by ambragoanta**
@DanceaRadu @Serban681 we need to make some changes in the autocomplete component

### CODE REVIEW

File: src/app/components/user-groups/group-form/group-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      users: formValue.users ?? [],
+      profilePicture:
+        this.uploadedPhotoId !== undefined
+          ? this.uploadedPhotoId
+          : this.initialGroup?.profilePicture,
+    };
+    this.formSubmit.emit(updatedGroup);
+  }
+
+  ngOnInit(): void {
```
Comment (DrBaxR) [inline src/app/components/user-groups/group-form/group-form.component.ts:99]: Please move ngoninit above ngonchanges

File: src/app/components/user-groups/group-form/group-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          .subscribe((photoId) => {
+            this.uploadedPhotoId = photoId;
+          });
+      } else {
+        console.error('User ID is undefined');
+      }
+    }
+  }
+
+  displayProfileImage(blob: Blob) {
```
Comment (DrBaxR) [inline src/app/components/user-groups/group-form/group-form.component.ts:152]: Use same extracted method here

File: src/app/components/user-groups/group/group.component.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      this.group?.profilePicture &&
+      this.profileCircle
+    ) {
+      this.groupService.getPhoto(this.group.userGroupId).subscribe((blob) => {
+        this.displayGroupImage(blob);
+      });
+    }
+  }
+
+  private displayGroupImage(blob: Blob) {
```
Comment (DrBaxR) [inline src/app/components/user-groups/group/group.component.ts:39]: Use same extracted method

File: src/app/components/user-groups/groups/groups.component.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  selector: 'app-groups',
+  templateUrl: './groups.component.html',
+  styleUrls: ['./groups.component.scss'],
+})
+export class GroupsComponent implements OnInit {
+  groups: FullUserGroupModel[] = [];
+  constructor(public groupService: GroupService) {}
+
+  ngOnInit(): void {
+    this.groupService.getAll().subscribe((data) => {
```
Comment (DrBaxR) [inline src/app/components/user-groups/groups/groups.component.ts:15]: use the `async` pipe instead

File: src/app/components/user-profile-form/user-profile-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+          .subscribe((blob) => {
+            this.displayProfileImage(blob);
+          });
+      } else {
+        console.error('User profile or profile picture is undefined.');
+      }
+    }
+  }
+
+  displayProfileImage(blob: Blob) {
```
Comment (DrBaxR) [inline src/app/components/user-profile-form/user-profile-form.component.ts:53]: use extracted method

File: src/app/service/user-profile.service.ts [hunk 6]
```diff
@@ -1,57 +1,124 @@
...
-    this.getByKeycloakId(userKeycloakId).subscribe((existingProfile) => {
-      let userProfile = existingProfile;
+  private userIdSubject = new BehaviorSubject<number | null>(null);
+  userId$ = this.userIdSubject.asObservable();
+  userId?: number;
 
-      if (userProfile.username == null) {
-        userProfile.username = userProfileKeycloak?.username;
+  userProfile: UserProfile | null = null;
+  userProfileUrl = 'http://localhost:8080/api/v1/user-profile';
```
Comment (DrBaxR) [inline src/app/service/user-profile.service.ts:23]: extract base url to environment file/use it here if it already exists

**Review by DrBaxR — CHANGES_REQUESTED** [review]

**Review by DrBaxR — APPROVED** [review]

**Review by DanceaRadu — APPROVED** [review]

**Pull request merged**